model_analyzer/model_manager.py

```diff
@@ -30,7 +29,7 @@ class ModelManager:
     """
 
     def __init__(self, config, gpus, client, server, metrics_manager,
-                 result_manager, state_manager):
+                 result_manager, state_manager, constraint_manager):
```

Comment:
It would be nice to boyscout in type checking along with these edits

Reply:
Added type checking for modified methods